agent/doctor/healthcheck.go

```diff
@@ -0,0 +1,67 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//      http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package doctor
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/dockerclient/dockerapi"
+	"github.com/cihub/seelog"
+)
+
+type Healthcheck interface {
+	GetHealthcheckStatus() HealthcheckStatus
+	GetHealthcheckTime() time.Time
+	RunCheck() bool
+}
+
+type dockerRuntimeHealthcheck struct {
```

Comment:
will this need to be persisted?

Reply:
Across Agent restarts?

Reply:
Although I added a LastCheckStatus/Time, we don't currently have a use for it.  And the only reason I can see persistance being necessary is to track our Last state.  

As we've talked about in team discussions -- these stats are really about the very latest stat.
So in case we start up after an agent reset (whether by service restart or instance reboot) and the doctor prematurely sends an aggreagate healthcheck message, the worst case would be that we'd see our defaults in cloudwatch.
```
                LatestCheckStatus: HealthcheckStatusInitializing,
                LatestCheckTime:   time.Now(),
```
Also note that the aggregation and message sending isn't part of the present PR but it should take into account the requirement of always initializing healthchecks where necessary.

So long story short: I don't think persistence is necessary.  Yet.